src/pg_tmgr.c

```diff
@@ -0,0 +1,110 @@
+#include "postgres.h"
+#include "libpq-fe.h"
+
+#include "connection.h"
+#include "pg_tmgr.h"
+
+#define MAX_STMT_LEN 1024
+
+int PgShardCurrTransManager;
+
+static bool PgShardBeginStub(PGconn* conn);
+static bool PgShardPrepareStub(PGconn* conn, char const* relationName, int64 shardId);
+static bool PgShardCommitPreparedStub(PGconn* conn, char const* relationName, int64 shardId);
+static bool PgShardRollbackPreparedStub(PGconn* conn, char const* relationName, int64 shardId);
+static bool PgShardRollbackStub(PGconn* conn);
+
+static bool PgShardBegin2PC(PGconn* conn);
+static bool PgShardPrepare2PC(PGconn* conn, char const* relationName, int64 shardId);
+static bool PgShardCommitPrepared2PC(PGconn* conn, char const* relationName, int64 shardId);
+static bool PgShardRollbackPrepared2PC(PGconn* conn, char const* relationName, int64 shardId);
+static bool PgShardRollback2PC(PGconn* conn);
+
+PgShardTransactionManager const PgShardTransManagerImpl[] = 
+{
+	{ PgShardBeginStub, PgShardPrepareStub, PgShardCommitPreparedStub, PgShardRollbackPreparedStub, PgShardRollbackStub },
+	{ PgShardBegin2PC, PgShardPrepare2PC, PgShardCommitPrepared2PC, PgShardRollbackPrepared2PC, PgShardRollback2PC }
+};
+
+/* 
+ * Transaction manager stub
+ */
+static bool PgShardBeginStub(PGconn* conn) 
```

Comment:
For all functions: the name of the function should be on a new line (like postgres source code)